### PR TITLE
Initial Integration

### PR DESCRIPTION
**Pull request by ghost**
This pull request is for the initial integration of GUI, Socket, and Game logic code on master branch. 
- Refactored Start, Connection, and Lobby screens
- Integrated the Game Logic 

Able to Launch GUI accept user input and display user name on the lobby screen.

### CODE REVIEW

File: src/gui/playerChoiceSpot.java [hunk 1]
```diff
@@ -3,11 +3,11 @@
 import javafx.scene.paint.Color;
 import javafx.scene.shape.Rectangle;
 
-public class playerChoiceSpot extends Rectangle {
-	
+public class PlayerChoiceSpot extends Rectangle
```
Comment (ryanlpeterson) [inline src/gui/playerChoiceSpot.java:6]: The filename also needs changed to PlayerChoiceSpot.

File: src/gui/LobbyScreen.java [hunk 2]
```diff
@@ -1,34 +1,78 @@
 package gui;
 
 import java.util.ArrayList;
-import java.util.Arrays;
```
Comment (ryanlpeterson) [inline src/gui/LobbyScreen.java]: Why was this import removed? Looks like it's still needed for line #42 in new file (within LobbyScreen(int assignedPlayerIndex, String[] allPlayerNames) constructor.

File: src/socket/ClientMessageReceiver.java [hunk 3]
```diff
@@ -126,10 +130,32 @@ public static void sendMessage(Object msg)
...
     {
-    	GameProcessor.processMessage(msg);
+    	if(object instanceof PlayerStatusMessage)
+    	{
+    		PlayerStatusMessage msg = (PlayerStatusMessage) object;
+    		
+    		processStatusMessage(msg);
+    	}
+
+      GameProcessor.processMessage(msg);
```
Comment (ryanlpeterson) [inline src/socket/ClientMessageReceiver.java]: Variable msg is not within scope here.

File: src/socket/ClientMessageReceiver.java [hunk 4]
```diff
@@ -126,10 +130,32 @@ public static void sendMessage(Object msg)
      * TODO: Game logic function should be called here. Function that receives Message and processes it.
      * @param msg
```
Comment (ryanlpeterson) [inline src/socket/ClientMessageReceiver.java]: Javadoc should be adjusted to reflect parameter name change.

File: src/controller/ViewController.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
+package controller;
+
+import logic.GameProcessor;
+import gui.UserInterface;
+
+public class ViewController
```
Comment (ryanlpeterson) [inline src/controller/ViewController.java]: Should we consider ditching this ViewController class? The idea would be for it to abstract/encapsulate the communication between the user's actions and the GameProcessor, but it seems that in most cases the ViewController won't really be encapsulating anything. The current flow would be: UI event handler -> ViewController -> GameProcessor, but the calls to GameProcessor seem like they will be like 1 line in most cases. Should we consider adjusting to: UI event handler -> GameProcessor, and consider our event handlers the "controller" in our MVC pattern?

**Review by ryanlpeterson — CHANGES_REQUESTED** [review]
I put some comments related to errors I saw when testing. Once these are resolved, I'll approve. The comment I made about the ViewController class could be discussed later if we'd rather merge sooner, but I think it should be discussed soon.

File: src/controller/ViewController.java (hunk 5 above)
Comment (ghost) [reply src/controller/ViewController.java]: Yeah, I like that idea. We can go that route

**Review by ghost — COMMENTED** [review]
Made all appropriate changes

**Review by ryanlpeterson — APPROVED** [review]
